packages/wasi/src/index.ts

```diff
@@ -732,6 +840,28 @@ export default class WASIDefault {
       fd_write: wrap(
         (fd: number, iovs: number, iovsLen: number, nwritten: number) => {
           const stats = CHECK_FD(fd, WASI_RIGHT_FD_WRITE);
+          if (this.asyncified) {
+            return asyncify(async () => {
+              let written = 0;
+              for (const iov of getiovs(iovs, iovsLen)) {
+                let w = 0;
+                while (w < iov.byteLength) {
+                  const i = await promisify(fs.write)(
```

Comment:
I think it would be better to use the `writeSync` API rather than make it awaitable and wait for the result.

```suggestion
                  const i = fs.writeSync(
```